backend/gsr_booking/api_wrapper.py

```diff
@@ -489,38 +230,76 @@ def book_room(self, rid, start, end, user, test=False):
-        )
-        group_gsrs = GSRBooking.objects.filter(
-            reservation__in=reservations, gsr__in=GSR.objects.filter(kind=GSR.KIND_LIBCAL)
-        )
+    def cancel_room(self, booking_id, user):
+        """Cancels room"""
+        # gsr_booking = get_object_or_404(GSRBooking, booking_id=booking_id)
+        # if gsr_booking:
+        #     if user != gsr_booking.user and user != gsr_booking.reservation.creator:
+        #         raise APIError("Error: Unauthorized: This reservation was booked by someone else.")
+        #     gsr_booking.is_cancelled = True
+        #     gsr_booking.save()
+        response = self.request("POST", f"{API_URL}/1.1/space/cancel/{booking_id}").json()
+        if "error" in response[0]:
+            raise APIError("LibCal: " + response[0]["error"])
+        return response
+    
+    def get_availability(self, gid, start, end, user):
+        """Returns a list of rooms and their availabilities"""
 
-        return GSRBookingSerializer(
-            GSRBooking.objects.filter(
-                user=user,
-                gsr__in=GSR.objects.filter(kind=GSR.KIND_LIBCAL),
-                end__gte=timezone.localtime(),
-                is_cancelled=False,
-            ).union(group_gsrs),
-            many=True,
-        ).data
+        # adjusts url based on start and end times
+        range_str = "availability"
+        if start:
+            start_datetime = datetime.datetime.combine(
+                datetime.datetime.strptime(start, "%Y-%m-%d").date(), datetime.datetime.min.time()
+            )
+            range_str += "=" + start
+            if end and not start == end:
+                range_str += "," + end
+        else:
+            start_datetime = None
+
+        # filters categories and then gets extra information on each room
+        
+        response = self.request("GET", f"{API_URL}/1.1/space/category/{gid}").json()
+        items = response[0]["items"]
+        items = ",".join([str(x) for x in items])
+        response = self.request("GET", f"{API_URL}/1.1/space/item/{items}?{range_str}")
 
+        if response.status_code != 200:
+            raise APIError(f"GSR Reserve: Error {response.status_code} when reserving data")
+        
+        rooms = [{"room_name": room['name'], 'id': room['id'], 'availability': room['availability']} for room in response.json() if room["id"] not in ROOM_BLACKLIST]
+        for room in rooms:
+            # remove extra fields
+            if "formid" in room:
+                del room["formid"]
+            # enforce date filter
+            # API returns dates outside of the range, fix this manually
+            room["availability"] = [
+                {
+                    'start_time': time['from'],
+                    'end_time': time['to']
+                }
+                for time in room["availability"]
+                if not start_datetime or datetime.datetime.strptime(time["from"][:-6], "%Y-%m-%dT%H:%M:%S") >= start_datetime
```

Comment:
Is there a reason we did the `[:-6]`, like why couldn't we do the whole `time["from"]` and add an extra `%z` at the end of the date formatter string

Reply:
idk this was based off of the logic that was here previously